### PR TITLE
M-4: Switch from DataAnnotations to EntityTypeConfigurations

### PR DESCRIPTION
**Pull request by darkness-reveur**

### CODE REVIEW

File: MeetupPlatformApi/Context/EntitiesConfiguration/MeetupEntityConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+﻿using MeetupPlatformApi.Entities;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+
+namespace MeetupPlatformApi.Context.EntitiesConfiguration
```
Comment (Rickstaper) [inline MeetupPlatformApi/Context/EntitiesConfiguration/MeetupEntityConfiguration.cs:5]: Use file scoped namespaces

File: MeetupPlatformApi/Context/EntitiesConfiguration/UserEntityConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
+﻿using MeetupPlatformApi.Entities;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+
+namespace MeetupPlatformApi.Context.EntitiesConfiguration
```
Comment (Rickstaper) [inline MeetupPlatformApi/Context/EntitiesConfiguration/UserEntityConfiguration.cs:5]: Use file scoped namespaces as well

**Review by Rickstaper — CHANGES_REQUESTED** [review]

File: MeetupPlatformApi/Context/EntitiesConfiguration/UserEntityConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+                .IsRequired()
+                .HasColumnName("pk_users");
+
+            builder.HasIndex(u => u.Username)
+                .HasName("ux_users_username")
+                .IsUnique();
+
+            builder.Property(u => u.Username)
+                .IsRequired()
+                .HasColumnName("ux_users_username");
```
Comment (satma0745) [inline MeetupPlatformApi/Context/EntitiesConfiguration/UserEntityConfiguration.cs:26]: Why would You use index name convention for a column?

**Review by Rickstaper — CHANGES_REQUESTED** [review]

File: MeetupPlatformApi/Context/EntitiesConfiguration/MeetupEntityConfiguration.cs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        meetupEntity.ToTable("meetups");
+
+        meetupEntity
+            .HasKey(meetup => meetup.Id)
+            .HasName("pk_meetups");
+
+        meetupEntity
+            .Property(meetup => meetup.Id)
+            .IsRequired()
+            .HasColumnName("pk_meetups");
```
Comment (satma0745) [inline MeetupPlatformApi/Context/EntitiesConfiguration/MeetupEntityConfiguration.cs:20]: Why have You named `Id` column `pk_meetups`?

File: MeetupPlatformApi/Context/EntitiesConfiguration/UserEntityConfiguration.cs [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        userEntity.ToTable("users");
+
+        userEntity
+            .HasKey(user => user.Id)
+            .HasName("pk_users");
+
+        userEntity
+            .Property(user => user.Id)
+            .IsRequired()
+            .HasColumnName("pk_users");
```
Comment (satma0745) [inline MeetupPlatformApi/Context/EntitiesConfiguration/UserEntityConfiguration.cs:20]: Why have You named `Id` column `pk_users`?

File: MeetupPlatformApi/Controllers/AuthenticationController.cs [hunk 6]
```diff
@@ -30,13 +30,21 @@ public async Task<IActionResult> GetUserById(Guid id)
     {
         var user = await context.Users.SingleOrDefaultAsync(user => user.Id == id);
         var outputDto = mapper.Map<UserOutputDto>(user);
-        return outputDto is null ? Ok(outputDto) : NotFound();
+        return outputDto is not null ? Ok(outputDto) : NotFound();
     }
 
     [HttpPost]
     public async Task<IActionResult> RegisterUser([FromBody] UserRegistrationDto userFromBody)
```
Comment (satma0745) [inline MeetupPlatformApi/Controllers/AuthenticationController.cs:37]: Please, rename DTO object to `registrationDto`

**Review by satma0745 — CHANGES_REQUESTED** [review]

File: MeetupPlatformApi/Controllers/AuthenticationController.cs [hunk 7]
```diff
@@ -30,13 +30,21 @@ public async Task<IActionResult> GetUserById(Guid id)
...
     }
 
     [HttpPost]
-    public async Task<IActionResult> RegisterUser([FromBody] UserRegistrationDto userFromBody)
+    public async Task<IActionResult> RegisterUser([FromBody] UserRegistrationDto registrationDto)
     {
-        var user = mapper.Map<UserEntity>(userFromBody);
+        var user = mapper.Map<UserEntity>(registrationDto);
+
+        var existingUser = await context.Users.AnyAsync(exUser => exUser.Username == user.Username);
```
[the remark this answers is not in the document]
Comment (satma0745) [reply MeetupPlatformApi/Controllers/AuthenticationController.cs:41]: Please, do **not** use abbreviations in variable/parameter names (`exUser`)

**Review by satma0745 — CHANGES_REQUESTED** [review]

**Review by satma0745 — APPROVED** [review]

**Review by Rickstaper — APPROVED** [review]

File: MeetupPlatformApi/Controllers/AuthenticationController.cs [hunk 8]
```diff
@@ -30,13 +30,21 @@ public async Task<IActionResult> GetUserById(Guid id)
...
-        return outputDto is null ? Ok(outputDto) : NotFound();
+        return outputDto is not null ? Ok(outputDto) : NotFound();
     }
 
     [HttpPost]
-    public async Task<IActionResult> RegisterUser([FromBody] UserRegistrationDto userFromBody)
+    public async Task<IActionResult> RegisterUser([FromBody] UserRegistrationDto registrationDto)
     {
-        var user = mapper.Map<UserEntity>(userFromBody);
+        var usernameAlreadyTaken = await context.Users.AnyAsync(user => user.Username == user.Username);
```
Comment (Rickstaper) [inline MeetupPlatformApi/Controllers/AuthenticationController.cs:39]: You're comparing identical object instances.
Change one of them from user.Username to registrationDto.Username

**Review by Rickstaper — CHANGES_REQUESTED** [review]

**Review by Rickstaper — APPROVED** [review]

**Pull request merged**